src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -674,87 +675,93 @@ private Index HandleStaticIndexChange(string name, IndexDefinition definition)
-                switch (definition.SourceType)
-                {
-                    case IndexSourceType.Documents:
-                        switch (definition.Type)
-                        {
-                            case IndexType.Map:
-                            case IndexType.JavaScriptMap:
-                                index = MapIndex.CreateNew(definition, _documentDatabase);
-                                break;
-
-                            case IndexType.MapReduce:
-                            case IndexType.JavaScriptMapReduce:
-                                var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceIndex>(definition, _documentDatabase);
-
-                                if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete.Count > 0)
-                                    mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
-
-                                index = mapReduceIndex;
-                                break;
-
-                            default:
-                                throw new NotSupportedException($"Cannot create {definition.Type} index from IndexDefinition");
-                        }
-
-                        break;
-
-                    case IndexSourceType.TimeSeries:
-                        switch (definition.Type)
-                        {
-                            case IndexType.Map:
-                            case IndexType.JavaScriptMap:
-                                index = MapTimeSeriesIndex.CreateNew(definition, _documentDatabase);
-                                break;
-
-                            case IndexType.MapReduce:
-                            case IndexType.JavaScriptMapReduce:
-                                var mapReduceIndex = MapReduceIndex.CreateNew<MapReduceTimeSeriesIndex>(definition, _documentDatabase);
-
-                                if (mapReduceIndex.OutputReduceToCollection != null && prefixesOfDocumentsToDelete.Count > 0)
-                                    mapReduceIndex.OutputReduceToCollection.AddPrefixesOfDocumentsToDelete(prefixesOfDocumentsToDelete);
-
-                                index = mapReduceIndex;
-                                break;
-
-                            default:
-                                throw new NotSupportedException($"Cannot create {definition.Type} index from TimeSeriesIndexDefinition");
-                        }
+                return CreateIndexFromDefinition(definition, _documentDatabase, prefixesOfDocumentsToDelete: prefixesOfDocumentsToDelete);
+            }
+        }
 
-                        break;
+        public Index CreateTestIndexFromDefinition(IndexDefinition indexDefinition, DocumentDatabase documentDatabase, DocumentsOperationContext context)
+        {
+            if (indexDefinition.Type == IndexType.JavaScriptMapReduce || indexDefinition.Type == IndexType.MapReduce)
```

Comment:
I think you can do `indexDefinition.Type.IsMapReduce()`

Reply:
Changed